ek9000App/src/devEL2XXX.cpp

```diff
@@ -69,16 +69,18 @@ static void EL20XX_WriteCallback(CALLBACK* callback) {
 	free(callback);
 
 	/* Check for invalid */
-	if (!dpvt->terminal)
+	if (!dpvt->terminal) {
+	        pRecord->pact = FALSE;
 		return;
+	}
 
 	/* Lock & verify mutex */
 	int status = dpvt->device->Lock();
 
 	if (status != epicsMutexLockOK) {
```

Comment:
Isn't `device->Lock` going to `asynPortDriver::lock` which returns `asynSuccess` or `asynError`? (Fortunately both success values are equal, in either case...)

Reply:
Good catch- originally it was using a standard epics mutex, but I changed that sometime in 2019 iirc (gosh time flies)